pkg/client/aad.go

```diff
@@ -19,41 +20,63 @@ type AadClient interface {
 	GetAadToken(ctx context.Context, clientID, clientSecret, tenantID, resource string) (string, error)
 }
 
-func NewAadClient(logger *logrus.Logger) AadClient {
+func NewAadClient(logger *logrus.Logger, aquireFunc aadTokenAcquirerFunc) AadClient {
 	return &aadClientImpl{
-		Logger: logger,
+		Logger:            logger,
+		tokenAcquirerFunc: aquireFunc,
 	}
 }
 
-type aadClientImpl struct {
-	Logger *logrus.Logger
+type TokenAcquirer interface {
+	Acquire(ctx context.Context, scopes []string) (confidential.AuthResult, error)
 }
 
-func (c *aadClientImpl) GetAadToken(ctx context.Context, clientID, clientSecret, tenantID, resource string) (string, error) {
-	scopes := []string{
-		fmt.Sprintf("%s/.default", resource),
-	}
+type aadTokenAcquirer struct {
```

Comment:
we shouldn't need any extra interfaces/structs with this change I don't think

Reply:
basically, we should only need the new custom type:

`type aadTokenAcquirerFunc func(authority, clientID, clientSecret string) (confidential.AuthResult, error)`

then a new field on `aadClientImpl` of type `aadTokenAcquirerFunc`

in NewAADClient(), I wouldn't expose the acquirerFunc has an argument, probably just always set it to the function that uses the real implementation - something like func `getAADTokenWithSecret(authority, clientID, clientSecret string) (confidential.AuthResult, error)`.

then in tests you can just construct raw `&aadClientImpl{}`'s and do something like:

`aadClient.aadTokenAcquirerFunc = func(authority, clientID, clientSecret string) (confidential.AuthResult, error) {...what ever we need in here...}`